src/server/cluster/cluster_config.cc

```diff
@@ -25,29 +27,58 @@ std::string_view ClusterConfig::KeyTag(std::string_view key) {
   return key.substr(start + 1, end - start - 1);
 }
 
-SlotId ClusterConfig::KeySlot(std::string_view key) {
-  std::string_view tag = KeyTag(key);
+SlotId ClusterConfig::KeySlot(string_view key) {
+  string_view tag = KeyTag(key);
   return crc16(tag.data(), tag.length()) & kMaxSlotNum;
 }
 
-ClusterConfig::ClusterConfig() {
+ClusterConfig::ClusterConfig(string_view my_id) : my_id_(my_id) {
   cluster_enabled = true;
-  AddSlots();
 }
 
-void ClusterConfig::AddSlots() {
-  // TODO update logic acording to config
-  // currently add all slots to owned slots
-  std::lock_guard lk{slots_mu_};
-  for (SlotId slot_id = 0; slot_id <= kMaxSlotNum; ++slot_id) {
-    owned_slots_.emplace(slot_id);
+namespace {
+optional<ClusterConfig::Node> GetMasterNodeForSlot(const vector<ClusterConfig::Node>& nodes,
+                                                   const string& my_id) {
+  for (const auto& node : nodes) {
+    if (node.id == my_id) {
+      // We do not store information for our own node, as it's unneeded, and it also helps us to
+      // determine our ownership quicker.
+      return nullopt;
+    }
+
+    if (node.role == ClusterConfig::Role::kMaster) {
+      return node;
+    }
+  }
+
+  CHECK(false) << "No master present in nodes";
```

Comment:
we dont want to crash the server if there is problem with the config data. You were thinking to add validation on this on a higher level later?

Reply:
I'll remove the `CHECK()` here, and will also add enforcement on `SetConfig()`

Reply:
I think its ok to have the CHECK here as an internal system behavior. But the SetConfig should have a return value so if the validations fail we should reply to the set config command with error